lib/code_corps/messages/emails.ex

```diff
@@ -0,0 +1,93 @@
+defmodule CodeCorps.Messages.Emails do
+  @moduledoc """
+  Handles email notifications used within the Messages context
+  """
+  alias CodeCorps.{
+    Conversation,
+    ConversationPart,
+    Emails,
+    Mailer,
+    Message,
+    Repo
+  }
+
+  @message_preloads [:project, [conversations: :user]]
+
+  @doc ~S"""
+  Notifies all relevant users of a new `CodeCorps.Message` being created.
+
+  We match against admin initiated messages only, because it's unclear who
+  message.author is for user-initiated messages, or who the email recipient
+  would be, assuming the message.author is the user who initiated the message.
+  """
+  @spec notify_of_new_message(Message.t) :: :ok
+  def notify_of_new_message(%Message{initiated_by: "admin"} = message) do
+    message = message |> Repo.preload(@message_preloads)
+
+    message
+    |> Map.get(:conversations)
+    |> Enum.map(&Emails.MessageForUserEmail.create(message, &1))
+    |> Enum.each(&Mailer.deliver_now/1)
+  end
+
+  @part_preloads [
+    conversation: [
+      [conversation_parts: :author],
+      [message: [:author, [project: :organization]]],
+      :user
+    ]
+  ]
```

Comment:
Again, ugly, same as above.

In this case, I might actually prefer doing an explicit sql query instead.